### PR TITLE
=htc migrate BodyPartRenderer to GraphStage #20288

### PR DESCRIPTION
**Pull request by leachbj**

### CODE REVIEW

File: akka-http-core/src/test/scala/akka/http/scaladsl/model/MultipartSpec.scala [hunk 1]
```diff
@@ -34,6 +35,16 @@ class MultipartSpec extends WordSpec with Matchers with Inside with BeforeAndAft
...
     }
+
+    "support `toEntity`" in {
+      val streamed = Multipart.General(
+        MediaTypes.`multipart/mixed`,
+        Source(Multipart.General.BodyPart(defaultEntity("data"), List(ETag("xzy"))) :: Nil))
+      val result = streamed.toEntity(boundary = "boundary")
+      result.contentType shouldBe MediaTypes.`multipart/mixed`.withBoundary("boundary").withCharset(HttpCharsets.`UTF-8`)
+      val encoding = Await.result(result.dataBytes.runWith(Sink.seq), 1.second)
+      encoding .map(_.utf8String).mkString shouldBe "--boundary\r\nContent-Type: text/plain; charset=UTF-8\r\nETag: \"xzy\"\r\n\r\ndata\r\n--boundary--"
```
Comment (leachbj) [inline akka-http-core/src/test/scala/akka/http/scaladsl/model/MultipartSpec.scala]: any suggestions for better/more tests?

Comment (johanandren) [inline akka-http-core/src/test/scala/akka/http/scaladsl/model/MultipartSpec.scala]: This was missing test coverage before?

**Comment by johanandren** [comment]
OK TO TEST

**Comment by drewhk** [comment]
LGTM as far as I can tell

**Comment by johanandren** [comment]
Binary compatibility check failed:

```
[error]  * method streamed(java.lang.String,java.nio.charset.Charset,Int,akka.event.LoggingAdapter)akka.stream.stage.PushPullStage in object akka.http.impl.engine.rendering.BodyPartRenderer has a different result type in current version, where it is akka.stream.stage.GraphStage rather than akka.stream.stage.PushPullStage
[error]    filter with: ProblemFilters.exclude[IncompatibleResultTypeProblem]("akka.http.impl.engine.rendering.BodyPartRenderer.streamed")
```

Should be safe to filter out by modifying `project/MiMa.scala` with the proposed filter since this is internal stuff.

**Comment by johanandren** [comment]
LGTM!

**Comment by johanandren** [comment]
Needs rebasing though, sorry about that. Busy days.

**Comment by leachbj** [comment]
@johanandren rebased again

**Comment by johanandren** [comment]
Thanks!

**Comment by johanandren** [comment]
Refs #20288

**Pull request merged**